dev-java/joda-time/joda-time-2.10.10-r1.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# Skeleton command:
+# java-ebuilder --generate-ebuild --workdir . --pom joda-time-2.10.10/pom.xml --download-uri https://github.com/JodaOrg/joda-time/archive/refs/tags/v2.10.10.tar.gz --slot 0 --keywords "~amd64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~sparc-solaris ~x86-solaris" --ebuild joda-time-2.10.10.ebuild
+
+EAPI=7
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="joda-time:joda-time:2.10.10"
+JAVA_TESTING_FRAMEWORKS="junit"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="Date and time library to replace JDK date handling"
+HOMEPAGE="https://www.joda.org/joda-time/"
+SRC_URI="https://github.com/JodaOrg/${PN}/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~sparc-solaris ~x86-solaris"
+
+# Common dependencies
+# POM: ${P}/pom.xml
+# org.joda:joda-convert:1.9.2 -> >=dev-java/joda-convert-2.2.1:0
+
+CDEPEND="
+	dev-java/joda-convert:0
+"
+
+DEPEND="
+	>=virtual/jdk-1.8:*
+	${CDEPEND}
+"
+
+RDEPEND="
+	>=virtual/jre-1.8:*
+	${CDEPEND}"
+
+S="${WORKDIR}"
+
+JAVA_GENTOO_CLASSPATH="joda-convert"
+JAVA_SRC_DIR="${P}/src/main/java"
+JAVA_RESOURCE_DIRS=(
+	"${P}/src/main/java"
+)
+
+JAVA_TEST_GENTOO_CLASSPATH="junit"
+JAVA_TEST_SRC_DIR="${P}/src/test/java"
+JAVA_TEST_RESOURCE_DIRS=(
+	"${P}/src/test/resources"
+)
+
+src_compile() {
+	java-pkg-simple_src_compile
+
+	ejavac -cp joda-time.jar org.joda.time.tz.ZoneInfoCompiler
```

Comment:
well, i think i was wrong with `ejavac`, in fact i suppose it should be quite similar to what you did with `hamcrest-core` in `src_compile`.

basically, you call `java`, specify the classpath using `-cp`, then the main class that should be run by `java`, and finally all the arguments that the main class accepts, like this:
```
"$(java-config -J)" \
                -cp ${PN}.jar \
                org.joda.time.tz.ZoneInfoCompiler \
                -src ... \
                -dst ... \
                africa \
                ...
```

Reply:
Yes, somewhat similar to the [`hamcrest-core`](https://github.com/gentoo/gentoo/blob/master/dev-java/hamcrest-core/hamcrest-core-1.3-r2.ebuild#L46-L55) experience.  But still some lessons to be learned. 
No `ZoneInfoMap.java`, no `ZoneInfoMap.class`. 
But finally it got `org/joda/time/tz/data/ZoneInfoMap` in the `.jar` file

[build.log.gz](https://github.com/gentoo/gentoo/files/6254466/build.log.gz)

`Closes: https://bugs.gentoo.org/779892` added
Does it really need the revbump? Or could it be fixed in-place?

Reply:
@vaukai it needs revbump, because it modifies content of the generated jar. without revbump, users would not get the updated/fixed jar

Reply:
Got it